ckanext/saml2/plugin.py

```diff
@@ -353,15 +391,23 @@ def _create_or_update_user(self, user_name, saml_info):
         context = {'schema': user_schema, 'ignore_auth': True}
 
         if is_new_user:
+            new_user_username = _get_random_username_from_email(
+                                            saml_info['email'][0])
+            name_id = saml_info['id'][0]
```

Comment:
I was thinking about ckanext-saml2 sometime over the weekend (probably while trying to go to sleep!) and wondered whether the stored NameID was coming from assertions/attributes or the NameID specified in the LoginResponse XML element (line 272). Ideally, it would use the NameId from the LoginResponse rather than an attribute that duplicates the NameID.

Now, for Data.NSW we already have an attribute that duplicates the NameID; I recall that we specified it and I'm trying to remember why.  It might have been to keep backwards compatibility with the original implementation at  https://github.com/GSA/ckanext-saml2. But I don't care so much about that anymore, GSA aren't interested in maintaining ckanext-saml2.

We should either change this or update the README to explain that the `saml2.user_mapping` must map the NameID to the `id` field.